File: Python/model/Igrac.py
from model.Entitet import Entitet


class Igrac(Entitet):
    def __init__(self, sifra=None, ime=None, prezime=None, url_slika=None, spol=None): # ne može id jer je to shadow build name
        super().__init__(sifra)
        if ime is None:
            self._ime = ""
        else:
            self._ime = ime
        self._prezime = "" if prezime is None else prezime
        self._url_slika = "" if url_slika is None else url_slika
        self._spol = "" if spol is None else spol

    @property
    def ime(self):
        return self._ime

    @ime.setter
    def ime(self, ime):
        self._ime = ime

    @property
    def prezime(self):
        return self._prezime

    @prezime.setter
    def prezime(self, prezime):
        self._prezime = prezime

    @property
    def url_slika(self):
        return self._url_slika

    @url_slika.setter
    def url_slika(self, url_slika):
        self._url_slika = url_slika

    @property
    def spol(self):
        return self._spol

    @spol.setter
    def spol(self, spol):
        self._spol = spol

    def __str__(self):
        return self.ime + " " + self.prezime

File: JS/JS/igrac.js
import { Entitet } from './entitet.js'
import { Spol } from './spol.js'

export class Igrac extends Entitet {
    #ime
    #prezime
    #urlSlika
    #spol

    constructor(id, ime, prezime, urlSlika, spol) {
        super(id)
        this.#ime = ime
        this.#prezime = prezime
        this.#urlSlika = urlSlika
        this.#spol = spol
    }

    getIme() {
        return this.#ime
    }

    setIme(ime) {
        this.#ime = ime
    }

    getPrezime() {
        return this.#prezime
    }

    setPrezime(prezime) {
        this.#prezime = prezime
    }

    getUrlSlika() {
        return this.#urlSlika
    }

    setUrlSlika(urlSlika) {
        this.#urlSlika = urlSlika
    }

    getSpol() {
        return this.#spol
    }

    setSpol(spol) {
        this.#spol = spol
    }

    getSpolNaziv() {
        return this.getSpol() === Spol.MUSKI ? 'Muško' : 'Žensko'
    }

    toString() {
        return `${this.getIme()} ${this.getPrezime()}`
    }
}